ca_driver/src/create_driver.cpp

```diff
@@ -222,6 +227,11 @@ void CreateDriver::undockCallback(const std_msgs::EmptyConstPtr& msg)
   robot_->setMode(create::MODE_FULL);
 }
 
+void CreateDriver::mainMotorCallback(const std_msgs::Float32ConstPtr& msg)
+{
+  robot_->setMainMotor(msg->data);
```

Comment:
Absolutely! What do you think of migrating most subscribers in the node to a service based model. Few of them really benefit from being topics (IMO just the velocity command subscriber). Other things like setting LEDs, requesting a song or turning off/on motors are just sporadically used and align better to the notion of service. What do you think about this? We also have the benefit of giving feedback to the requester via the service response if something went wrong (ex: the underlying driver failed for whatever reason). Maybe I can start with one and see if it really makes create_autonomy more flexible/powerful.